scraper/radio_scrape/radio_scrape/image_colour.py

```diff
@@ -1,107 +1,157 @@
-    hist /= hist.sum()
 
-    # Group cluster's (percentage, rgb, hex), 
-    colours = []
-    # define named tuple with fields for percent, and lab
-    ColourDetails = namedtuple('Colour', ('frequency', 'lab'))
-    for (percent, color) in zip(hist, centroids):
+def find_dominant_img_colours(image_path: str, quantity: int = 3) -> list[str]:
+    """
+        param image_path: string path to image file
+        param quantity: int number of colours to return
+        Returns: list with specified number of hex colours sorted from darkest to lightest. 
+        If the image does not contain enough colouts, the list will be shorter.
 
-        # convert color to colormath rgb so we can easily perfrom conversions
-        rgb = sRGBColor(*color, is_upscaled=True)
-        lab = convert_color(rgb, LabColor)
-        # print('rgb', rgb)
-        # print('lab', lab)
+        k-means clustering is used to cluster n similar colours together, and obtain the mean colour of each group.
+        n is specified by the quantity parameter.
 
-        # desaturate and lower contrast so bg colours don't compete with image
-        labProcessed = decontrast(desaturate(lab))
-        # print('labProcessed', labProcessed)
+        These central colours of each cluster are then processed(de-saturated, with reduced dynamic range) to obtain results that are more suitable for use as a background gradient behing the image.
 
-        # add named tuple to colours list with percent, rgb, and hex
-        colours.append(ColourDetails(percent,labProcessed))
+        Colours are then filtered for uniqueness, sorted from darkest to lightest, and returned as a list of hex colours.
+    """
+    # Load image and convert to a list of pixels
+    image_data_RGB_2d = rgb_2d_pixel_array_from_image(image_path)
 
-    return colours
+    # init kmeans - cluster the image data into the specified number of clusters
+    clusters = KMeans(n_clusters=quantity).fit(image_data_RGB_2d)
 
+    central_cluster_colours_and_frequencies_zip = group_cluster_colour_and_frequency(clusters)
 
-def desaturate(lab):
-    """
-        Desaturates a colour by converting it to hls and then back to lab.
-    """
+    central_cluster_colours = []
+    for (frequency, color) in central_cluster_colours_and_frequencies_zip:
+         central_cluster_colours.append(CentralClusterColour(frequency=frequency, orig_rgb = sRGBColor(*color, is_upscaled=True)))
 
-    # convert lab to hsl 
-    hsl = convert_color(lab, HSLColor)
+    colours_filtered_for_uniqueness = filter_for_uniqueness(central_cluster_colours)
+ 
+    # Sort from darkest to lightest based on lab L* value
+    colours_filtered_for_uniqueness.sort(reverse=False, key=lambda colour: colour.processed_lab.lab_l)
     
-    # desaturate a little more the more saturated it is
-    hsl.hsl_s = hsl.hsl_s - ((hsl.hsl_s * 0.25) + (hsl.hsl_s / 65) * 5)
+    filtered_lab_sorted_hex_colours = [colour.processed_hex for colour in colours_filtered_for_uniqueness]
```

Comment:
improve variable name. Don't think we need to mention that this list was sorted using LAB colour space.